### PR TITLE
test(utils): raise src/utils/formatting.ts to zero escaped mutants and put it back under the gate

### LINKED ISSUE #203: Raise the mutation score of src/utils/formatting.ts
Follow-up to #197, which introduced mutation testing and the changed-lines PR gate.

> **Status update (#197 as merged).** `src/utils/formatting.ts` is now **excluded from the `mutate` scope** — `'!src/utils/formatting.ts'` in `stryker.config.mjs`. That was necessary for #197 to ship self-contained: the gate fails a PR on a survivor in a line it changed, and the 42 lines below carry one, so the debt in this file would have failed unrelated PRs that merely touched it. Excluded from *mutation* only — the file's tests still run and still count for coverage, including the boundary tests #197 added.
>
> **Re-adding it to the scope is part of this issue**, and should be the last step, once the work below is done. Until then this file is unguarded by the gate, which is the cost of the deferral and the reason not to leave it open indefinitely.

## Why it matters, and not just as a number

The gate fails a PR on a mutant that survives **in a line the PR changed**. There are 42 distinct lines in this file carrying a surviving logic mutant, so with the file in scope, touching any of them failed CI and the author had to write the missing assertion as a side quest to whatever they were actually doing. That is why the file is excluded for now rather than gated — but the exclusion trades a blocked PR for no signal at all, so it is a deferral, not a resolution.

## Measured

Re-measured on `581c958` (2026-08-22), `pnpm test:mutation --mutate 'src/utils/formatting.ts'`, with the August figures for comparison:

| | 7f4edb2 (Aug) | `581c958` (now) |
| --- | ---: | ---: |
| Mutation score (total / covered) | 67.43 % / 72.84 % | **67.72 % / 73.04 %** |
| Mutants | 525 | 536 |
| Killed | 354 | 363 |
| Survived | 132 | 134 |
| NoCoverage | 39 | 39 |
| **Escaped (Survived + NoCoverage)** | **171** | **173** |

**Unchanged.** Every other area in #208's scope gained ground over those three weeks — the full scope went 77.19 % to 84.25 % — because the gate makes each PR pay for the lines it touches. This file is out of the `mutate` scope, so nothing has asked anything of it. That is the exclusion's real cost, measured rather than asserted.

Escaped, by mutator (`581c958`):

| Mutator | Count |
| --- | --- |
| StringLiteral | 102 (59 %) |
| ConditionalExpression | 27 |
| EqualityOperator | 13 |
| MethodExpression | 12 |
| ArrayDeclaration | 5 |
| LogicalOperator | 5 |
| ArrowFunction | 3 |
| BlockStatement | 2 |
| ArithmeticOperator | 2 |
| BooleanLiteral | 1 |
| OptionalChaining | 1 |

**71 of the 173 are not label strings.** That is the part worth stating plainly: the "it's mostly `.describe()`-style noise" reading — the argument that took `src/tools/**` out of scope in §5 of the ADR — is only 59 % true here. The rest is unasserted branching, spread over 42 lines:

| Escaped logic mutants | Location | Mutators |
| --- | --- | --- |
| 13 | `AUDIT_ENTITY_FIELDS` | BlockStatement, ArrayDeclaration, ConditionalExpression, EqualityOperator |
| 10 | `formatSlaPolicy` | MethodExpression, ArrowFunction, ConditionalExpression, EqualityOperator |
| 9 | `formatTagDiff` | ArrayDeclaration, MethodExpression, ArrowFunction, BooleanLiteral, ConditionalExpression, ArithmeticOperator |
| 7 | `formatMacro` | MethodExpression, ConditionalExpression, ArithmeticOperator, BlockStatement, EqualityOperator |
| 7 | `formatOrganization` | MethodExpression, ConditionalExpression, EqualityOperator |
| 6 | `formatTicketField` | ArrayDeclaration, MethodExpression, ConditionalExpression, EqualityOperator |
| 5 | `formatTicket` | MethodExpression, LogicalOperator, ConditionalExpression, EqualityOperator |
| 4 | `formatSlaBlock` | LogicalOperator, EqualityOperator, ConditionalExpression, MethodExpression |
| 3 | `formatCategory` | ConditionalExpression, LogicalOperator |
| 3 | `formatSection` | ConditionalExpression, LogicalOperator |
| 1 each | `formatAudit`, `formatUser`, `formatArticleSummary`, `formatList` | OptionalChaining, MethodExpression |

## The 71 logic mutants: not optional

These are real gaps — a `ConditionalExpression` survivor means the branch produces the same output either way as far as any test can tell. Write the assertions. No decision needed, just work.

## The label strings: decided

**Settled — see the [decision comment](https://github.com/fruggr/zendesk-mcp-server/issues/203#issuecomment-5382868266)** (it carries the evidence, the sampled survivors and the verified mechanism). Neither A nor B as this issue framed them, because the framing's premise does not hold: `StringLiteral` is not a proxy for "label prose". The policy is **per string role**, not per mutator or per file:

1. **Behavioural strings — assert, never disable.** Dispatch keys, `case` labels, sentinels, algorithm/encoding identifiers, HTTP header names and values, protocol parameters, error codes, data-mapping keys. In *this* file that includes `:272` (`value === ''`), `:288` (`field === 'tags'`), `:315` (`case 'VoiceComment'`) and `:228-248` (`AUDIT_ENTITY_FIELDS`).
2. **Output prose read by an agent — pin the high-traffic formatters** (`formatTicket`, `formatArticleSummary`, `formatOrganization`, `formatUser`) with exact-output assertions; **block-scoped disable for the long tail**, each with a reason.
3. **Internal diagnostics — disable with a reason.**

**Correction to the mechanism this issue recorded in August: a file-scoped disable is the wrong tool here.** It would have silently exempted the role-1 strings in the list above — the mutants most worth killing in this file. The mechanism is a **block**, verified on `src/utils/validation.ts`: `// Stryker disable StringLiteral: <reason>` … `// Stryker restore StringLiteral` marks only the enclosed lines `Ignored` (out of numerator and denominator, invisible to the gate), while `StringLiteral` mutants outside the block stay scored. `disable next-line` for a single one. Never at file scope, and `mutator.excludedMutations` stays unused — it is global-only, so it would exempt files nobody has looked at.

**So the classification is the work.** The 102 `StringLiteral` survivors split into role 1 (tests, non-negotiable), role 2 tail (block disable with a reason) and the high-traffic formatters (golden output). Do that pass before writing anything.

## Acceptance

- [ ] The 71 non-StringLiteral escaped mutants are killed, or individually justified with `// Stryker disable next-line <mutator>: <reason>` where genuinely equivalent
- [ ] The 102 `StringLiteral` survivors classified by role, and the role-1 ones (sentinels, dispatch keys, `AUDIT_ENTITY_FIELDS`) killed with tests — not disabled
- [ ] Exact-output assertions on the high-traffic formatters; block-scoped `// Stryker disable StringLiteral: <reason>` … `// Stryker restore StringLiteral` for the prose tail, each disable naming the role it claims
- [ ] `docs/decisions/mutation-testing.md` §5 rewritten to the role-based policy — it currently states the "mostly `.describe()` noise" argument, which the decision comment supersedes
- [ ] No `src/` behaviour changes — this is test work (same constraint as #197)
- [ ] `pnpm test:mutation --mutate 'src/utils/formatting.ts'` shows the new score, quoted in the PR
- [ ] **`'!src/utils/formatting.ts'` is removed from `mutate` in `stryker.config.mjs`**, putting the file back under the gate — and the ADR's §5 exclusion table updated to match. Do this last: `pnpm test:mutation:diff origin/main HEAD` must be green on the PR that does it, which is the proof the work is actually finished.

## Reproduce

```sh
pnpm test:mutation --mutate 'src/utils/formatting.ts'   # 536 mutants, ~4 min on 4 cores
```

Then, for the breakdown:

```sh
node -e '
const r=JSON.parse(require("fs").readFileSync("reports/mutation/mutation.json","utf8"));
const esc=r.files["src/utils/formatting.ts"].mutants.filter(m=>m.status==="Survived"||m.status==="NoCoverage");
const by={};for(const m of esc) by[m.mutatorName]=(by[m.mutatorName]??0)+1;
console.log(esc.length, Object.entries(by).sort((a,b)=>b[1]-a[1]));'
```

Background: [`docs/decisions/mutation-testing.md`](../blob/main/docs/decisions/mutation-testing.md).

### PR DESCRIPTION
**Pull request by dlecan**
## Summary

`src/utils/formatting.ts` has been out of the `mutate` scope since #197, at 67.72 % with **173 escaped mutants over 101 of its 461 lines**. This PR takes it to **100.00 % — 536 killed, 0 survived, 0 without coverage, 0 ignored** — and removes the exclusion, so the file is gated again.

| `src/utils/formatting.ts` | `5958501` (before) | this branch |
| --- | ---: | ---: |
| Mutation score (total / covered) | 67.72 % / 73.04 % | **100.00 % / 100.00 %** |
| Killed | 363 | 536 |
| Survived | 134 | **0** |
| NoCoverage | 39 | **0** |
| Ignored | 0 | **0** |
| Lines carrying an escaped mutant | 101 of 461 | **0** |

No Stryker directive anywhere in the file. The `src/` diff is comments only.

The bar is zero escaped, not a better score — see the [analysis comment](https://github.com/fruggr/zendesk-mcp-server/issues/203#issuecomment-5474063628). Re-arming the gate with a residue would just relocate the booby trap #197 deferred: a mine on each of those 101 lines for the next author who touches one.

Whole-scope side effects, measured: **2 043 mutants at 88.40 %** (was 1 452 documented), cold run **13 min 7 s** locally on 4 cores.

## Linked issue

Closes #203

## Type of change
- [ ] Bug fix
- [ ] New feature
- [ ] Refactor (no external behavior change)
- [x] Documentation
- [x] Tooling / CI

Primarily test work; the ADR rewrite, the `CONTRIBUTING.md` convention and the `stryker.config.mjs` scope flip come with it.

## Author checklist
- [x] I checked no open or merged PR already addresses the linked issue, and the issue is not already closed as completed / `released`
- [x] If this PR resolves an issue, its description above links it with a closing keyword so it auto-closes on merge to `main`
- [x] `pnpm test` passes locally — 954 tests
- [x] `pnpm test:coverage` meets the thresholds — 98.07 / 89.03 / 99.34 / 98.5 against 94 / 77 / 98 / 95
- [x] `pnpm check` is clean (lint + format)
- [x] `pnpm typecheck` passes
- [x] I have read the diff myself, line by line
- [x] I ran a Claude Code review on the diff and addressed its findings — see *What review caught* below; three review passes each found something real, and all are fixed or filed
- [x] Documentation is updated where needed
- [x] If this PR adds an MCP tool: n/a
- [x] If the change has **MCP-runtime-observable** behaviour: **exempt.** Tests, mutation-testing config and docs; the `src/` change is comments. A running server behaves identically for a client, so the standard gates are the validation.

### Why inline snapshots, and why no block-scoped disable

The root cause of the 173 was the assertion style, not the strings: this file's suite ran **116 `toContain` against 13 `toBe`**. `toContain('open')` on `formatTicket` says nothing about the template — emptying `- **Status**: ` leaves `open` in the output. Three mutant classes are invisible to `toContain` no matter how many keywords you add:

- an emptied separator (`', '`, or the trailing `.join('\n')`);
- a dropped `.filter(Boolean)` (the output gains a blank line);
- the **suppressed-line branch** of a ternary — Stryker rewrites `''` to a non-empty marker, so the line *appears* and `expect(result).not.toContain('Promoted')` still passes. 29 of the 102 `StringLiteral` survivors were this, and they are not prose.

One exact-output assertion kills the `StringLiteral`, `ConditionalExpression`, `EqualityOperator` and `MethodExpression` on the same line at once, which is why this was not split into "logic mutants" and "label strings".

Both treatments were measured on `formatOrganization` (`:358-366`, 21 mutants, baseline 6 killed / 15 escaped):

| | Score | Killed | Ignored | Survived |
| --- | ---: | ---: | ---: | ---: |
| two inline snapshots, no directive | **100.00 %** | 21 | 0 | **0** |
| `// Stryker disable StringLiteral` block, assertions unchanged | 36.36 % | 4 | 10 | **7** |

The 7 survivors under the directive are the logic mutants, and killing them needs *the same two inputs* the snapshots already use. The disable buys back no test work — it was costed against hand-written expected output, and a pre-filled `toMatchInlineSnapshot` reduces that to one `vitest` run. That is what ADR §5 now says, on the axis that actually decides it (what the string *does*, four roles) rather than per mutator or per file.

### What review caught, and the lesson that came out of it

An earlier revision waived two mutants in `renderAuditValue` as equivalent. **Both were killable.** CodeRabbit flagged the first; the second had the identical flaw and nothing had flagged it.

The flaw was where the argument lived, not how the waiver was scoped:

- `renderAuditValue` empties `''` before resolving an id because `withName` goes through `Number(value)`, and `Number('')` is 0 — so without the guard, a cleared field picks up whatever name sits at key 0. I justified the waiver with `addPositiveId` in `tools/tickets.ts` never putting 0 in those maps. That is a fact about today's caller; `AuditNames` is a plain `Map<number, string>` and promises nothing.
- `!Array.isArray(value)` stops an array being destructured for `minutes` and reported as a duration. I argued no input could separate the two because `JSON.parse` cannot produce an array with an own `minutes` property. Also a fact about the caller — `value` is `unknown`.

Each died to one test asserting the guard's own contract, so both directives are gone, and reverting the guard split that waiver-scoping had needed put `src/` back to comments only.

ADR §5 now carries the generalisable form in place of the waiver-scoping advice: **when a mutant looks equivalent, check whether the argument is a fact about the *type* or about *today's caller*. If it is the caller, the mutant is marking an unasserted contract, and the assertion is the fix.** The scoping limitation is still recorded — `disable next-line` cannot target one mutant among several of the same mutator on a line, and hoisting a condition into a named `const` does not narrow it because Stryker emits the whole-conjunction mutant wherever the expression sits — but as a last resort, after that question.

A later pass caught two more of mine: a test whose name endorsed the opposite of its assertion (`treats a deadline exactly at the current instant as due, not as remaining`, asserting a snapshot that reads `0 min remaining`), and a `CONTRIBUTING.md` bullet telling authors to fill an empty snapshot with `-u` while the bullet below it calls `-u` a product-surface change. Verified both directions before rewriting: a plain `pnpm test <file>` fills an empty inline snapshot and *fails* on a filled one that no longer matches, without overwriting.

### One defect found and deliberately not fixed here — #260

`formatSlaBlock` computes its `Next breach` line over every metric with a future `breach_at` and no stage filter, while `formatSlaMetric` ten lines below suppresses the countdown for `paused`, `achieved` and `fulfilled`. The two disagree inside the same block, and a snapshot added here shows three non-live metrics under a `Next breach` header. Pre-existing, but this PR pins it — so the snapshot carries a comment saying it records today's output rather than endorsed behaviour, and #260 owns the fix plus the product call on whether `paused` should feed the header. Fixing it changes user-visible output, which is a different PR.

### The diff gate is not the proof, and this PR demonstrates it

`pnpm test:mutation:diff origin/main HEAD` on the commit that re-arms the gate:

```
Mutating the changed lines: src/utils/formatting.ts:272-277,src/utils/formatting.ts:282-284
Stryker produced no mutants inside the changed lines, so there is nothing to
gate. Note this is not a pass: ...
```

Zero mutants judged — the script says so itself. `changedRanges` reads `git diff --unified=0` and `escapedMutants` judges only a mutant contained in a changed range, and the only `formatting.ts` lines this PR changes are comments. The gate would have reported exactly the same with 173 survivors still in the file. The full-file run is the proof; the gate is a smoke check. Corrected in the issue's acceptance list in a comment.

### Also

- **30 tests removed**, each strictly subsumed by an exact-output snapshot over the identical input in the same describe. Tests whose name states a rule the output does not show on its own — a comment body must never leak into the timeline — are kept. Mutation testing was re-run after the deletions and still reports 536/536, which is the proof they carried no signal.
- **The SLA block moved onto fake timers.** A real clock makes "some number of minutes" the loosest possible claim about arithmetic. A pinned `now` states every branch exactly, including two boundaries that were unreachable before: a deadline exactly at the current instant is not a *future* breach, and 0 minutes remaining is not 0 minutes overdue.
- **`tests/unit/mutation-scope.test.ts` needs no change** — its `'!src/utils/formatting.ts'` patterns are literal fixtures for `scopeMatcher`, not read from the config. The `!` paragraph stays in ADR §5 for the same reason: no negation is left in `mutate` today, but the semantics still have to hold for the next exclusion.
- **Coverage thresholds untouched.** Branches now sit 12 points above the 77 floor, but that gap mostly predates this PR and raising a global floor affects unrelated work — a ratchet belongs in its own change.

https://claude.ai/code/session_01X3S7XFhsXkVFmeiHLdTANT

## Summary by CodeRabbit

* **Documentation**
  * Added guidance for writing and maintaining committed inline snapshots.
  * Updated mutation-testing documentation and scope details.

* **Tests**
  * Expanded formatting coverage across tickets, users, organizations, SLAs, audits, translations, and other supported values.
  * Added snapshot-based assertions and edge-case coverage, including malformed timestamps and missing data.

* **Bug Fixes**
  * Preserved formatting behavior while improving handling of empty and missing values.

### CODE REVIEW

File: src/utils/formatting.ts [hunk 1]
```diff
@@ -269,10 +269,36 @@ const withName = (id: unknown, names: Map<number, string>): string => {
...
+  // early-out, not a sentinel the output depends on: every path further down
+  // already yields '' for '' — formatFieldValue via String(''), and withName via
+  // Number('') === 0, whose lookup cannot hit (the maps are built by
+  // `addPositiveId` in `tools/tickets.ts`, which admits positive integers only, so
+  // key 0 is absent by construction). No input can observe its removal, so skipping
+  // the early-out (ConditionalExpression => false, BlockStatement) and never
+  // matching it (StringLiteral) are equivalent. The waiver also takes one killed
+  // sibling with it, "always return ''", which the audit suite still fails on; the
+  // returned '' on the line below stays scored.
+  // Stryker disable next-line ConditionalExpression,StringLiteral,BlockStatement: equivalent, see above
```
[the remark this answers is not in the document]
Comment (dlecan) [reply src/utils/formatting.ts]: Correct, and it invalidated more than the one waiver. Fixed in cd8d31e.

The flaw was where my argument lived. I justified the waiver with `addPositiveId` in `tools/tickets.ts` never putting 0 in those maps — a fact about today's caller, not about the function. `AuditNames` is a plain `Map<number, string>` and promises nothing, so the guard is the contract and the mutant was marking it unasserted. One test with key 0 in the map kills it.

The same flaw was in the second waiver, which nothing had flagged: `!Array.isArray(value)` stops an array being destructured for `minutes` and reported as a duration, and I argued no input could separate the two because `JSON.parse` cannot produce an array with an own `minutes` property. Also a fact about the caller — `value` is `unknown`. One test with `Object.assign(['1', '2'], { minutes: 5 })` kills that one.

So both directives are gone and the file needs none:

```
File           |  total | covered | # killed | # timeout | # survived | # no cov |
 formatting.ts | 100.00 |  100.00 |      536 |         0 |          0 |        0 |
```

536 mutants, 536 killed, **0 ignored**. This also let me revert the guard split the waiver-scoping needed, so the `src/` diff is back to comments only — which is what #203 asked for.

Two notes on your suggested options. Narrowing the `AuditNames` contract would work but costs a branded type and its ripple; asserting the guard is cheaper and documents why the guard exists, which the code did not say. And the generalisable form is now in ADR §5, replacing the waiver-scoping advice it carried: when a mutant looks equivalent, check whether the argument is a fact about the *type* or about *today's caller* — if it is the caller, the mutant is marking an unasserted contract, and the assertion is the fix.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by dlecan** [comment]
Two review passes landed in `ba3f751`. Score unchanged at **536/536, 100 %**, which is what makes the deletions safe to claim.

**Seven assertions removed**, each a strict substring subset of an exact-output snapshot over the identical input in the same describe: `formatUser`'s two null-field negatives, four `formatArticleSummary` cases, and `formatMacro`'s substring check. Re-running mutation testing after the deletions is the proof they carried no signal of their own.

**One test name was endorsing the opposite of its assertion.** `treats a deadline exactly at the current instant as due, not as remaining` asserts a snapshot reading `0 min remaining`. The boundary it actually pins is the next-breach one (`t > Date.now()`), so a maintainer flipping `remaining < 0` to `<= 0` would have read the name as backing them. Renamed, and the comment now separates the two halves.

**A `CONTRIBUTING.md` bullet contradicted the one below it.** I had told authors to fill an empty inline snapshot with `pnpm exec vitest run <file> -u`, while the next bullet calls `-u` a product-surface change needing justification. Verified both directions before rewriting it: a plain `pnpm test <file>` fills an empty inline snapshot, and on a *filled* one that no longer matches it fails without overwriting. So the recipe is a plain run, with `-u` reserved for deliberately re-baselining.

### One finding deliberately not fixed here — #260

Review also surfaced a real defect in production output that this PR does not introduce but does **pin**. `formatSlaBlock` computes its `Next breach` line over every metric with a future `breach_at` and no stage filter, while `formatSlaMetric` ten lines below suppresses the countdown for `paused`, `achieved` and `fulfilled`. The two disagree inside the same block, and the snapshot added here shows it:

```
### SLA
- **Next breach**: 2026-06-01T13:00:00.000Z
- **agent_work_time** — paused; due 2026-06-01T13:00:00.000Z
- **first_reply_time** — achieved; due 2026-06-01T13:00:00.000Z
- **total_resolution_time** — fulfilled; due 2026-06-01T13:00:00.000Z
```

Three metrics with no live obligation, one breach announced an hour out. For the agent reading this, `Next breach` is the line it acts on.

Not fixed in this PR: it changes user-visible formatter output, and it needs a product call on whether `paused` should feed the header (`achieved` and `fulfilled` are clear-cut; a paused SLA can resume). That is scope discipline, not a punt — #260 carries the analysis, the two options and a recommendation. The snapshot now carries a comment saying it pins today's output rather than endorsed behaviour, so it cannot silently bless the bug.

Worth noting for whoever takes #260: the fix lands on lines now under the gate at 100 %, so it will need a per-stage assertion rather than an updated snapshot — which is the gate doing its job on the first change after re-arming.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: docs/decisions/mutation-testing.md [hunk 2]
```diff
@@ -362,22 +366,109 @@ scope is a decision to do its assertion work *first*:
...
+**And it cannot target one mutant among several of the same mutator on a line.**
+The equivalent mutant is typically the whole-condition `ConditionalExpression`
+while its siblings on the same line are killed, so a line-level waiver takes them
+with it. Splitting a guard into two statements isolates a clause; hoisting a
+condition into a named `const` does *not*, because Stryker emits the
+whole-conjunction mutant wherever the expression sits. Which is why the first
+question is never "how do I scope the waiver" but **"is this mutant actually
+equivalent"**.
+
+#203 got that wrong twice before getting it right, and the pattern is worth
```
[the remark this answers is not in the document]
Comment (dlecan) [reply docs/decisions/mutation-testing.md]: Fixed in c204e76 — `Issue #203 got that wrong…`, and it was the only line-initial `#<digits>` in the repo's docs, so nothing else is affected.

Took the committable suggestion rather than the prose one: `Issue \`#203\`` would suppress GitHub's autolink to the issue, and the reference is more useful clickable than fenced. The MD018 ambiguity is gone either way, since the hash is no longer at the start of the line.

Worth noting the finding is about other Markdown renderers, not this one — GitHub needs a space after `#` to make a heading, so it was already rendering as prose here. Still worth fixing rather than arguing: the linter is right that a bare hash at line start is ambiguous, and the sentence reads better with the prefix.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by dlecan** [comment]
`main` moved to `07be788` (lockfile maintenance, #261) and the PR went `behind`. Merged in and revalidated at `8903b52`:

- `pnpm install --frozen-lockfile`, `pnpm check`, `pnpm typecheck`, `pnpm build` — clean
- 954 tests
- `formatting.ts`: **536 mutants, 536 killed, 100.00 %** — holds across the merge
- all 8 CI checks green (`Full scope` skipped, as it is on every PR)

### On the `Linked Issues` pre-merge warning

CodeRabbit's latest review flags a real ambiguity rather than a defect: #203's acceptance criterion says `pnpm test:mutation:diff origin/main HEAD` must be green and calls that *the proof the work is finished*, while this PR's description says the gate reports "this is not a pass". Both of its suggested resolutions are satisfied:

**The gate does not fail.** The `Changed lines` job is **success** on every head of this PR, `8903b52` included, and the script exits 0. Its "this is not a pass" line is advisory about *interpretation*, printed next to a successful exit: it means "green here proves little", not "this failed". Nothing in the gate needs changing — line granularity is deliberate and documented in ADR §4, and a gate that judged a whole file on any edit would recreate the booby trap #203 exists to remove.

**The criterion is amended**, since the clause it hinged on cannot verify what it claims: [#203 comment](https://github.com/fruggr/zendesk-mcp-server/issues/203#issuecomment-5489622774). It now reads as *the changed-lines gate must not fail, and the full-file run must show zero escaped mutants — the latter is the proof.* Both are met.

---
_Generated by [Claude Code](https://claude.ai/code)_
